docs/tutorials/rest_api/index.md

```diff
@@ -0,0 +1,405 @@
+   part.id = price.part AND \n\
+   vendor.id = price.vendor;\"
+}"
+```
+
+(The many escape characters in the command are to both have the SQL readable enough
+and be correct JSON simultaneously.)
+
+As response, we should get back `HTTP/1.1 201 Created` along with
+the identifier of the program and its version (1).
+When an SQL program is created or when its code is updated, its version is incremented
+and compilation is automatically triggered.
+
+Check that the program is compiled:
+
+```
+curl -X GET http://localhost:8080/v0/programs/sc-program
+```
+
+... which will have `status` set to `Success` when compilation is done.
+
+### Step 2: Data connectors
+
+Data connectors are used to both input data sources into stream tables,
+and output data from incremental views to data sinks. In this tutorial,
+we will define an HTTP source for each of the input stream tables:
+
+* **Part (named `sc-connector-part`):**
+  ```
+  curl -i -X PUT http://localhost:8080/v0/connectors/sc-connector-part \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "description": "Connector for part",
+    "config": {
+        "transport": {
+            "name": "url",
+            "config": {
+                "path": "https://feldera-basics-tutorial.s3.amazonaws.com/part.json"
+            }
+        },
+        "format": {
+            "name": "json",
+            "config": {}
+        }
+    }
+  }'
+  ```
+
+* **Vendor (named `sc-connector-vendor`):**
+  ```
+  curl -i -X PUT http://localhost:8080/v0/connectors/sc-connector-vendor \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "description": "Connector for vendor",
+    "config": {
+        "transport": {
+            "name": "url",
+            "config": {
+                "path": "https://feldera-basics-tutorial.s3.amazonaws.com/vendor.json"
+            }
+        },
+        "format": {
+            "name": "json",
+            "config": {}
+        }
+    }
+  }'
+  ```
+
+* **Price (named `sc-connector-price`):**
+  ```
+  curl -i -X PUT http://localhost:8080/v0/connectors/sc-connector-price \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "description": "Connector for price",
+    "config": {
+        "transport": {
+            "name": "url",
+            "config": {
+                "path": "https://feldera-basics-tutorial.s3.amazonaws.com/price.json"
+            }
+        },
+        "format": {
+            "name": "json",
+            "config": {}
+        }
+    }
+  }'
+  ```
+
+Each of the above should return `HTTP/1.1 201 Created` when successful.
+
+We can view the existing connectors using:
+
+```
+curl -X GET http://localhost:8080/v0/connectors
+```
+
+Because data sinks require write permission (rather than just read),
+we do not create output connectors in this tutorial. However, the procedure
+to do so is the same, albeit of course with its own specific configuration
+(e.g., providing output topic in a Kafka output connector config).
```

Comment:
This paragraph isn't very helpful. Let's just say that we don't create sink connectors (and maybe refer to a section of the tutorial under construction) and that we are going to read outputs via the API instead.

Reply:
I changed it to a note:

```
> In this tutorial we do not create output connectors. We will read the
> views via the API (see step 6). The procedure to create output connectors
> is the same as for input connectors, albeit of course with its own
> specific configuration (e.g., providing output topic in a Kafka output
> connector config).
```